Yoast/YoastSEO.js
# Refactor urlkeyword assessment

## Summary

This PR can be summarized in the following changelog entry:

* Implements a new scoring schema for the assessment that checks if the keyword is used in the slug. If the keyword has 1 or 2 content words: GOOD if all content words are in the URL, OK otherwise. If the keyword has >2 content words: GOOD if > half of content words are in the URL, OK otherwise.
* Fixes the bug when the keyword would not be found in the slug when containing punctuation.

## Relevant technical choices:

* Also fixed the missing quotation marks in the KeyphraseDistribution assessment.

## Test instructions

This PR can be tested by following these steps:

* Make a post with a short keyphrase (2 content words) and make sure to not use them in the slug. The assessment should return an OK result.
* Use one of the words from your keyphrase in your slug. The assessment should still return an OK result.
* Use the second word from the keyphrase in your slug. The assessment should return a GOOD result.
* Add 3 more content words to your keyphrase without changing the slug. The assessment should return an OK result.
* Use one of the words from your keyphrase in your slug. The assessment should return a GOOD result.
* Remove the keyphrase. The assessment should not appear at all.
* Since this PR also fixes a couple bugs with keyphrase recognition in slugs, try using punctuation marks in the keyphrase (e.g., `St. Paul's Cathedral - Visit`, `pooch & mutt`).

Fixes https://github.com/Yoast/YoastSEO.js/issues/1650 Fixes #1608 Fixes #278
